lib/_stream_readable.js

```diff
@@ -256,7 +256,8 @@ function readableAddChunk(stream, chunk, encoding, addToFront) {
     } else if (state.objectMode || chunk && chunk.length > 0) {
       if (typeof chunk !== 'string' &&
           !state.objectMode &&
-          Object.getPrototypeOf(chunk) !== Buffer.prototype) {
+          // Do not use Object.getPrototypeOf as it is slower.
+          !(chunk instanceof Buffer)) {
```

Comment:
This is not exactly the same as before but I guess that's not important in this case. The former check was stricter.

Reply:
Yes I know. However it was not covered by tests, so it was definitely not intended.